lib/dentaku/type/checker.rb

```diff
@@ -2,6 +2,7 @@ module Dentaku
   module Type
     class Checker
       class UnboundIdentifier < StandardError
+        attr_reader :identifier
```

Comment:
Introducing this means that there's a shadowing in `resolve_identifier`

Reply:
Different class - this is the error class contained within `Checker`. There's only three methods including `initialize` :]

(Also this whole class gets refactored and moved to its own file in #12)